internal/controllers/topology/cluster/desired_state.go

```diff
@@ -81,18 +81,12 @@ func (r *Reconciler) computeDesiredState(ctx context.Context, s *scope.Scope) (*
 	// - Building the TopologyReconciled condition.
 	// - Make upgrade decisions on the control plane.
 	// - Making upgrade decisions on machine pools.
-	if len(s.Current.MachinePools) > 0 {
```

Comment:
I think this was required to make `clusterctl alpha topology plan` work. Why are we removing it?

Reply:
It looks like `r.Tracker` is nil when any of the cluster_controller tests are run. Removing this fixed that but the problem is definitely somewhere else, so that's my mistake! Will take a closer look right now.

Reply:
Yeah I think we have to add a custom tracker here which returns a fake client.